frontends/web/src/routes/account/add/components/coin-dropdown.tsx

```diff
@@ -16,39 +17,100 @@
 
 import { useTranslation } from 'react-i18next';
 import * as backendAPI from '@/api/backend';
-import { Select } from '@/components/forms';
+import Select, { components, SingleValueProps, OptionProps, SingleValue, DropdownIndicatorProps } from 'react-select';
+import { Logo } from '@/components/icon/logo';
+import styles from './coin-dropdown.module.css';
 
 type TCoinDropDownProps = {
   onChange: (coin: backendAPI.ICoin) => void;
   supportedCoins: backendAPI.ICoin[];
   value: string;
 };
 
+type TOption = {
+  label: string;
+  value: backendAPI.ICoin['coinCode'];
+  isDisabled: boolean;
+}
+
+
+const DropdownIndicator = (props: DropdownIndicatorProps<TOption>) => {
+  return (
+    <components.DropdownIndicator {...props}>
+      <div className={styles.dropdown} />
+    </components.DropdownIndicator>
+  );
+};
+
+const SelectSingleValue = (props: SingleValueProps<TOption>) => {
+  const { value, label } = props.data;
+  return (
+    <div className={styles.singleValueContainer}>
+      <components.SingleValue {...props}>
+        <div className={styles.valueContainer}>
+          <Logo coinCode={value} alt={label} />
+          <span className={styles.selectLabelText}>{label}</span>
+        </div>
+      </components.SingleValue>
+    </div>
+  );
+};
+
+
+const SelectOption = (props: OptionProps<TOption>) => {
+  const { label, value } = props.data;
+
+  return (
+    <components.Option {...props}>
+      <div className={styles.valueContainer}>
+        <Logo coinCode={value} alt={label} />
+        <span className={styles.selectLabelText}>{label}</span>
+      </div>
+    </components.Option>
+  );
+};
+
+
 export const CoinDropDown = ({
   onChange,
   supportedCoins,
   value,
 }: TCoinDropDownProps) => {
   const { t } = useTranslation();
+
+  const options = [
+    { label: t('buy.info.selectPlaceholder'), value: 'choose', isDisabled: true },
+    ...supportedCoins.map(({ coinCode, name, canAddAccount }) => ({
+      value: coinCode,
+      label: name,
+      isDisabled: !canAddAccount,
+    })),
+  ] as TOption[];
```

Comment:
I get this error:

<img width="1017" alt="Screenshot 2024-11-12 at 07 44 08" src="https://github.com/user-attachments/assets/51a5b6ea-2ba0-457e-a7c8-12dfdd0921ca">

Reply:
better not cast to TOption here. Something like this fixes the error:

```diff
diff --git a/frontends/web/src/routes/account/add/components/coin-dropdown.tsx b/frontends/web/src/routes/account/add/components/coin-dropdown.tsx
index eda2c8691..ddcbd012d 100644
--- a/frontends/web/src/routes/account/add/components/coin-dropdown.tsx
+++ b/frontends/web/src/routes/account/add/components/coin-dropdown.tsx
@@ -31,7 +31,11 @@ type TOption = {
   label: string;
   value: backendAPI.ICoin['coinCode'];
   isDisabled: boolean;
-}
+} | {
+  label: string;
+  value: 'choose';
+  isDisabled: true;
+};
 
 
 const DropdownIndicator = (props: DropdownIndicatorProps<TOption>) => {
@@ -48,7 +52,9 @@ const SelectSingleValue = (props: SingleValueProps<TOption>) => {
     <div className={styles.singleValueContainer}>
       <components.SingleValue {...props}>
         <div className={styles.valueContainer}>
-          <Logo coinCode={value} alt={label} />
+          {value !== 'choose' && (
+            <Logo coinCode={value} alt={label} />
+          )}
           <span className={styles.selectLabelText}>{label}</span>
         </div>
       </components.SingleValue>
@@ -63,7 +69,9 @@ const SelectOption = (props: OptionProps<TOption>) => {
   return (
     <components.Option {...props}>
       <div className={styles.valueContainer}>
-        <Logo coinCode={value} alt={label} />
+        {value !== 'choose' && (
+          <Logo coinCode={value} alt={label} />
+        )}
         <span className={styles.selectLabelText}>{label}</span>
       </div>
     </components.Option>
@@ -78,14 +86,14 @@ export const CoinDropDown = ({
 }: TCoinDropDownProps) => {
   const { t } = useTranslation();
 
-  const options = [
+  const options: TOption[] = [
     { label: t('buy.info.selectPlaceholder'), value: 'choose', isDisabled: true },
     ...supportedCoins.map(({ coinCode, name, canAddAccount }) => ({
       value: coinCode,
       label: name,
       isDisabled: !canAddAccount,
     })),
-  ] as TOption[];
+  ];
 
   return (
     <Select
```